controllers/events/eventsource_controller.go

```diff
@@ -390,64 +378,42 @@ func (r *EventSourceReconciler) handleEventSource(eventSource *openfunctionevent
 		if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, component, mutateDaprComponent(r.Scheme, component, eventSource)); err != nil {
 			return err
 		}
+		r.controlledResources.SetResourceStatus(component.Name, ResourceTypeComponent, Running)
 
 		// Create the workload for EventSource.
-		if _, err := r.createOrUpdateEventSourceWorkload(eventSource, ss.SourceKind, ss.EventName, workloadsPendingDelete); err != nil {
+		if _, err := r.createOrUpdateEventSourceWorkload(eventSource, ss.SourceKind, ss.EventName); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (r *EventSourceReconciler) handleDeprecatedComponents(eventSource *openfunctionevent.EventSource, cpd map[string]bool) error {
-	log := r.Log.WithName("handleDeprecatedComponents")
-	var component componentsv1alpha1.Component
-	for componentName, isDeprecated := range cpd {
-		if isDeprecated {
-			err := r.Get(r.ctx, types.NamespacedName{Namespace: eventSource.Namespace, Name: componentName}, &component)
-			if err != nil {
-				if util.IsNotFound(err) {
-					continue
+func (r *EventSourceReconciler) handleDeprecatedResources(eventSource *openfunctionevent.EventSource) error {
+	log := r.Log.WithName("handleDeprecatedResources")
 
-				} else {
-					log.Error(err, "Failed to get deprecated component", "namespace", eventSource.Namespace, "name", componentName)
-					return err
-				}
-			}
-			if err := r.Delete(r.ctx, &component); err != nil {
-				log.Error(err, "Failed to delete deprecated component", "namespace", eventSource.Namespace, "name", componentName)
+	// handle deprecated workloads
+	for workloadName, workloadStatus := range r.controlledResources.Workloads {
+		if workloadStatus.IsDeprecated {
+			if err := r.Delete(r.ctx, workloadStatus.Object); util.IgnoreNotFound(err) != nil {
+				log.Error(err, "Failed to delete deprecated workload", "namespace", eventSource.Namespace, "name", workloadName)
```

Comment:
workload is deleted, do we need to delete corresponding items in r.controlledResources.Workloads ?

Reply:
In the statistics, I filtered out the deleted workloads based on the "IsDeprecated" property.
It has no matter at the moment, but I will consider whether we need to delete it later.

Reply:
> In the statistics, I filtered out the deleted workloads based on the "IsDeprecated" property.
> It has no matter at the moment, but I will consider whether we need to delete it later.

if items in r.controlledResources.Workloads are not deleted, it will keep growing and will not be garbage collected.
So it's better to clear these obsolete resources